src/components/Note/Note.jsx

```diff
@@ -4,34 +4,94 @@ import {
   BiArchiveIn,
   BiTrashAlt,
   BsPin,
+  BsPinFill,
   MdOutlineLabel,
+  MdClose,
 } from "../../utils/icons/icons";
+import LabelButton from "../LabelButton/LabelButton";
 import "./note.css";
 
 function Note() {
+  const defaultState = {
+    title: "",
+    body: "",
+    isPinned: false,
+    tags: ["home", "fruits"],
+    bgColor: "#ffffff",
+    createdAt: new Date().toLocaleDateString(),
+  };
+
+  const [noteData, setNoteData] = useState(defaultState);
+  console.log(noteData);
+
+  const removetag = (_item) => {
+    setNoteData({
+      ...noteData,
+      tags: noteData.tags.filter((item) => item !== _item),
+    });
+  };
+
   return (
-    <div className="note flex-column">
-      <BsPin className="note-pin text-lg" />
-      <div className="note-content flex-column">
-        <input
-          className="note-title p-xs"
-          type="text"
-          placeholder="Title of the note"
-        />
-        <textarea
-          className="note-body p-xs"
-          type="text"
-          rows={5}
-          placeholder="Body of the note"
-        />
+    <div className="note " style={{ backgroundColor: `${noteData.bgColor}` }}>
+      <div className="flex-column">
+        <button
+          className="note-pin text-lg icon-button"
+          onClick={() =>
+            setNoteData({ ...noteData, isPinned: !noteData.isPinned })
+          }
+        >
+          {noteData.isPinned ? <BsPinFill /> : <BsPin />}
+        </button>
+
+        <div className="note-content flex-column">
+          <textarea
+            className="note-title p-xs"
+            type="text"
+            placeholder="Title of the note"
+            value={noteData.title}
+            onChange={(e) =>
+              setNoteData({ ...noteData, title: e.target.value })
+            }
+            rows="1"
+          />
+          <textarea
+            className="note-body p-xs"
+            type="text"
+            placeholder="Body of the note"
+            value={noteData.body}
+            onChange={(e) => setNoteData({ ...noteData, body: e.target.value })}
+          />
```

Comment:
will suggest to write timer for long text onChange input  as the function will get called less times.

Reply:
If we create a timer function to update the value in noteData then nothing will be rendered in that area for that span of time, as the textarea is a controlled input which is reading its value from noteData